### PR TITLE
[ Xedra Evolved ] Species based snares [Robot, Cyborg, Nether, Changeling]

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Content "Species Based Snare traps"

#### Purpose of change
I wanted to add snare traps that capture robots and cyborgs.  Renech made that possible.  This PR is the result.

#### Describe the solution
Uses Renech's code to add species based snares for Robot and cyborg snares.  

#### Describe alternatives you've considered
Might add hedge magic and alchemy snares.

#### Testing
All complete

### CODE REVIEW

File: data/mods/Xedra_Evolved/items/inventor/traps.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    "flags": [ "SINGLE_USE" ],
+    "use_action": {
+      "type": "place_trap",
+      "trap": "tr_species_snare",
+      "moves": 150,
+      "practice": 2,
+      "done_message": "You set the qr code."
+    }
+  }
+]
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/inventor/traps.json:50]:
```suggestion
]

```

File: data/mods/Xedra_Evolved/items/inventor/traps.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
+[
+  {
+    "id": "stunning_qr_trap",
+    "type": "TOOL",
+    "name": "stunning qr trap",
```
Comment (Qrox) [inline data/mods/Xedra_Evolved/items/inventor/traps.json:5]:
```suggestion
    "name": "stunning QR trap",
```

File: data/mods/Xedra_Evolved/items/inventor/traps.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    "material": "plastic",
+    "symbol": "*",
+    "color": "light_gray",
+    "flags": [ "SINGLE_USE" ],
+    "use_action": {
+      "type": "place_trap",
+      "trap": "tr_species_snare",
+      "moves": 150,
+      "practice": 2,
+      "done_message": "You set the qr code."
```
Comment (Qrox) [inline data/mods/Xedra_Evolved/items/inventor/traps.json:23]:
```suggestion
      "done_message": "You set the QR code."
```

File: data/mods/Xedra_Evolved/items/inventor/traps.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
```
Comment (Qrox) [inline data/mods/Xedra_Evolved/items/inventor/traps.json:47]:
```suggestion
      "done_message": "You set the QR code."
```

File: data/mods/Xedra_Evolved/items/alchemy.json [hunk 5]
```diff
@@ -113,6 +113,30 @@
...
     }
   },
+  {
+    "id": "nether_snare",
+    "type": "TOOL",
+    "name": "snaring pentagram trap",
+    "category": "traps",
+    "properties": { "capture_species": "NETHER" },
+    "description": "A metallic pentagram full of mystic symbols.  Will trap creatures from other dimensions if they wander into it.",
+    "weight": "1 Kg",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/alchemy.json:123]:
```suggestion
    "weight": "1 kg",
```

File: data/mods/Xedra_Evolved/items/alchemy.json [hunk 6]
```diff
@@ -113,6 +113,30 @@
...
+  {
+    "id": "nether_snare",
+    "type": "TOOL",
+    "name": "snaring pentagram trap",
+    "category": "traps",
+    "properties": { "capture_species": "NETHER" },
+    "description": "A metallic pentagram full of mystic symbols.  Will trap creatures from other dimensions if they wander into it.",
+    "weight": "1 kg",
+    "volume": "1000 ml",
+    "longest_side": "1 m",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/alchemy.json:125]:
```suggestion
    "longest_side": "60 cm",
```

File: src/monmove.cpp [hunk 7]
```diff
@@ -65,6 +65,7 @@ static const efftype_id effect_dragging( "dragging" );
 static const efftype_id effect_grabbed( "grabbed" );
 static const efftype_id effect_harnessed( "harnessed" );
 static const efftype_id effect_led_by_leash( "led_by_leash" );
+static const efftype_id effect_immobilization( "immobilization" );
```
Comment (RenechCDDA) [inline src/monmove.cpp:68]: Need to move this up one line

Comment (RenechCDDA) [inline src/monmove.cpp:68]:
```suggestion
static const efftype_id effect_immobilization( "immobilization" );
static const efftype_id effect_led_by_leash( "led_by_leash" );
```

Oops OK, github's suggestion UI is finnicky. Let's try this.

File: data/mods/Xedra_Evolved/recipes/alchemy.json [hunk 8]
```diff
@@ -192,6 +192,20 @@
...
+    "result": "nether_snare",
+    "category": "CC_XEDRA",
+    "subcategory": "CSC_XEDRA_ALCHEMY",
+    "skill_used": "fabrication",
+    "skills_required": [ "deduction", 1 ],
+    "difficulty": 3,
+    "time": "75 m",
+    "flags": [ "SECRET" ],
+    "tools": [ [ [ "paint_brush", -1 ] ] ],
+    "components": [ [ [ "paint", 1, "LIST" ] ], [ [ "scrap_dreamdross", 3 ] ], [ [ "mutant_blood", 1 ] ], [ [ "silver_small", 1 ] ] ]
```
Comment (RenechCDDA) [inline data/mods/Xedra_Evolved/recipes/alchemy.json:207]:
```suggestion
    "tools": [ [ [ "paint_brush", -1 ] ], [ [ "paint", 1, "LIST" ] ] ],
    "components": [ [ [ "scrap_dreamdross", 3 ] ], [ [ "mutant_blood", 1 ] ], [ [ "silver_small", 1 ] ] ]
```

Resolves the load error and makes the recipe actually craftable

File: src/trapfunc.cpp [hunk 9]
```diff
@@ -53,6 +53,7 @@ static const damage_type_id damage_pure( "pure" );
 static const efftype_id effect_beartrap( "beartrap" );
 static const efftype_id effect_heavysnare( "heavysnare" );
 static const efftype_id effect_in_pit( "in_pit" );
+static const efftype_id effect_immobilization( "immobilization" );
```
Comment (RenechCDDA) [inline src/trapfunc.cpp:56]:
```suggestion
static const efftype_id effect_immobilization( "immobilization" );
static const efftype_id effect_in_pit( "in_pit" );
```

I am apparently terrible at the alphabet?? Fixes another clang error

File: src/trapfunc.cpp [hunk 10]
```diff
@@ -682,6 +683,67 @@ bool trapfunc::snare_heavy( const tripoint &p, Creature *c, item * )
...
+        const bodypart_id hit = critter->get_random_body_part_of_type( body_part_type::type::leg );
+
+        critter->add_msg_if_player( m_bad, _( "A %1$s catches your %2$s!" ),
+                                    laid_trap.name(), hit->name.translated() );
+
+        if( critter->has_effect( effect_ridden ) ) {
+            add_msg( m_warning, _( "Your %1$s is caught by a %2$s!" ), critter->get_name(), laid_trap.name() );
+        }
+        if( !critter->is_avatar() ) {
+            add_msg_if_player_sees( p, "%1$s is caught by a %2$s!", critter->get_name(), laid_trap.name() );
```
Comment (RenechCDDA) [inline src/trapfunc.cpp:736]:
```suggestion
            add_msg_if_player_sees( p, _( "%1$s is caught by a %2$s!" ), critter->get_name(), laid_trap.name() );
```

**Comment by Maddremor** [comment]
Do these traps allow for capturing monsters/deactivate robots where applicable?

**Comment by RenechCDDA** [comment]
> Do these traps allow for capturing monsters/deactivate robots where applicable?

Any monsters with the correct species that fails the avoidance roll when stepping on the trap's tile should be captured. It disables them for 10 turns, but it does NOT turn robots into deactivated robots (which is technically their corpse).

File: data/json/effects.json [hunk 11]
```diff
@@ -623,6 +623,15 @@
     ],
     "flags": [ "EFFECT_LIMB_SCORE_MOD" ]
   },
+  {
+    "type": "effect_type",
```
Comment (Maleclypse) [inline data/json/effects.json:627]:
```suggestion
    "type": "effect_type"
```

File: data/json/effects.json [hunk 12]
```diff
@@ -623,6 +623,15 @@
     ],
     "flags": [ "EFFECT_LIMB_SCORE_MOD" ]
   },
+  {
+    "type": "effect_type"
```
Comment (Maleclypse) [inline data/json/effects.json:627]:
```suggestion
    "type": "effect_type",
```

File: data/mods/Xedra_Evolved/recipes/inventor/misc.json [hunk 13]
```diff
@@ -144,6 +144,54 @@
...
+    "qualities": [ { "id": "SCREW", "level": 1 }, { "id": "SAW_M", "level": 1 } ],
+    "components": [
+      [ [ "light_disposable_cell", 1 ] ],
+      [ [ "power_supply", 1 ] ],
+      [ [ "scrap", 3 ] ],
+      [ [ "cable", 4 ] ],
+      [ [ "scrap_dreamdross", 4 ] ],
+      [ [ "plastic_chunk", 2 ] ],
+      [ [ "paint", 2, "LIST" ] ],
+      [ [ "amplifier", 1 ] ]
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/recipes/inventor/misc.json:168]:
```suggestion
    "tools": [ [ [ "paint_brush", -1 ] ], [ [ "paint", 1, "LIST" ] ] ],
    "components": [
      [ [ "light_disposable_cell", 1 ] ],
      [ [ "power_supply", 1 ] ],
      [ [ "scrap", 3 ] ],
      [ [ "cable", 4 ] ],
      [ [ "scrap_dreamdross", 4 ] ],
      [ [ "plastic_chunk", 2 ] ],
      [ [ "amplifier", 1 ] ]
```

File: data/mods/Xedra_Evolved/recipes/inventor/misc.json [hunk 14]
```diff
@@ -144,6 +144,54 @@
...
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/recipes/inventor/misc.json:192]:
```suggestion
    "tools": [ [ [ "paint_brush", -1 ] ], [ [ "paint", 1, "LIST" ] ] ],
    "components": [
      [ [ "light_disposable_cell", 1 ] ],
      [ [ "power_supply", 1 ] ],
      [ [ "scrap", 3 ] ],
      [ [ "cable", 4 ] ],
      [ [ "scrap_dreamdross", 4 ] ],
      [ [ "plastic_chunk", 2 ] ],
      [ [ "amplifier", 1 ] ]
```

**Pull request merged**